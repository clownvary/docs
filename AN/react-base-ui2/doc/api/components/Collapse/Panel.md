## Panel 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
className|string &#124; object|&ensp;|false|customize class for Collapse section
Header|string &#124; number &#124; node|&ensp;|false|&ensp;
content|string &#124; number &#124; node|&ensp;|false|content of the panel
disabled|bool|&ensp;|false|If true, panel cannot be opened or closed