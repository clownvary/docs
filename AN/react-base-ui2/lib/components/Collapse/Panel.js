'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _defineProperty2 = require('babel-runtime/helpers/defineProperty');var _defineProperty3 = _interopRequireDefault(_defineProperty2);var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _propTypes = require('prop-types');var _propTypes2 = _interopRequireDefault(_propTypes);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);
var _consts = require('../../consts');
var _dom = require('../../utils/dom');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/**
                                                                                                                                     * Default PropTypes for Panel
                                                                                                                                     */
var propTypes = {
  /**
                   * customize class for Collapse section
                   */
  className: _propTypes2.default.oneOfType([
  _propTypes2.default.string,
  _propTypes2.default.object]),

  Header: _propTypes2.default.oneOfType([
  _propTypes2.default.string,
  _propTypes2.default.number,
  _propTypes2.default.node]),

  /**
                               * content of the panel
                               */
  content: _propTypes2.default.oneOfType([
  _propTypes2.default.string,
  _propTypes2.default.number,
  _propTypes2.default.node]),

  /**
                               * If true, panel cannot be opened or closed
                               */
  disabled: _propTypes2.default.bool,
  /**
                                       * Determines the aria-label text on expanded
                                       */
  ariaLableExpand: _propTypes2.default.string,
  /**
                                                * Determines the aria-label text on collapsed
                                                */
  ariaLableCollapse: _propTypes2.default.string };var


Panel = function (_React$Component) {(0, _inherits3.default)(Panel, _React$Component);function Panel() {var _ref;var _temp, _this, _ret;(0, _classCallCheck3.default)(this, Panel);for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {args[_key] = arguments[_key];}return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, (_ref = Panel.__proto__ || (0, _getPrototypeOf2.default)(Panel)).call.apply(_ref, [this].concat(args))), _this), _this.








    handleItemClick = function (e) {
      e && e.preventDefault();var
      onItemClick = _this.props.onItemClick;
      onItemClick();
    }, _this.





    onPanelHeaderKeyDown = function (e) {
      var keyCode = e.keyCode || e.which;var
      prefixCls = _this.props.prefixCls;
      var headerClass = prefixCls + '-item__header';
      var isEnterOrSpace = keyCode === _consts.KeyCode.ENTER || keyCode === _consts.KeyCode.SPACE;

      if (isEnterOrSpace) {
        if ((0, _dom.hasClass)(e.target, headerClass)) {
          _this.handleItemClick(e);
          return _this.props.onPanelHeaderKeyDown(e, keyCode);
        }
      }

      return _this.props.onPanelHeaderKeyDown(e, keyCode);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);}(0, _createClass3.default)(Panel, [{ key: 'handleHeaderClick', value: function handleHeaderClick(e) {e.stopPropagation();} }, { key: 'renderDefaultHeader', value: function renderDefaultHeader()

    {var
      Header = this.props.Header;

      return (
        _react2.default.createElement('div', { onClick: this.handleHeaderClick },
          Header,
          _react2.default.createElement('i', {
            className: 'icon icon-chevron-down',
            onClick: this.handleItemClick })));



    } }, { key: 'render', value: function render()

    {var _classNames, _classNames2;var _props =











      this.props,prefixCls = _props.prefixCls,Header = _props.Header,className = _props.className,content = _props.content,isActive = _props.isActive,disabled = _props.disabled,children = _props.children,ariaLableExpand = _props.ariaLableExpand,ariaLableCollapse = _props.ariaLableCollapse,isPanelHeaderFocusable = _props.isPanelHeaderFocusable;

      var panelCls = (0, _classnames2.default)((_classNames = {}, (0, _defineProperty3.default)(_classNames,
      prefixCls + '-item', true), (0, _defineProperty3.default)(_classNames,
      'is-expanded', isActive), _classNames),
      className);

      var headerCls = (0, _classnames2.default)((_classNames2 = {}, (0, _defineProperty3.default)(_classNames2,
      prefixCls + '-item__header', true), (0, _defineProperty3.default)(_classNames2,
      'is-disabled', disabled), (0, _defineProperty3.default)(_classNames2,
      'is-focusable', isPanelHeaderFocusable), _classNames2));

      return (
        _react2.default.createElement('div', { className: panelCls },
          _react2.default.createElement('a', {
              className: headerCls,
              'aria-disabled': disabled,
              tabIndex: disabled ? -1 : 0,
              href: 'javascript:void(0)' // eslint-disable-line
              , onClick: this.handleItemClick,
              onKeyDown: this.onPanelHeaderKeyDown,
              'aria-label': isActive ? ariaLableExpand : ariaLableCollapse,
              role: 'button',
              'aria-expanded': isActive },


            (0, _isFunction2.default)(Header) ?
            _react2.default.createElement(Header, { onToggleClick: this.handleItemClick }) :

            this.renderDefaultHeader()),



          _react2.default.createElement('div', {
              className: prefixCls + '-item__content' },

            children || content)));



    } }]);return Panel;}(_react2.default.Component);Panel.propTypes = propTypes;Panel.defaultProps = { headerButtonId: 'default', onItemClick: function onItemClick() {}, isActive: false };exports.default =


Panel;module.exports = exports['default'];