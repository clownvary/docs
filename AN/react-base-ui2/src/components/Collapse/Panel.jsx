import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import isFunction from 'lodash/isFunction';
import { KeyCode } from '../../consts';
import { hasClass } from '../../utils/dom';

/**
 * Default PropTypes for Panel
 */
const propTypes = {
  /**
   * customize class for Collapse section
   */
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  Header: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]),
  /**
   * content of the panel
   */
  content: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]),
  /**
   * If true, panel cannot be opened or closed
   */
  disabled: PropTypes.bool,
  /**
   * Determines the aria-label text on expanded
   */
  ariaLableExpand: PropTypes.string,
  /**
   * Determines the aria-label text on collapsed
   */
  ariaLableCollapse: PropTypes.string
};

class Panel extends React.Component {
  static propTypes = propTypes

  static defaultProps = {
    headerButtonId: 'default',
    onItemClick() { },
    isActive: false
  }

  handleItemClick = (e) => {
    e && e.preventDefault();
    const { onItemClick } = this.props;
    onItemClick();
  }

  handleHeaderClick(e) {
    e.stopPropagation();
  }

  onPanelHeaderKeyDown = (e) => {
    const keyCode = e.keyCode || e.which;
    const { prefixCls } = this.props;
    const headerClass = `${prefixCls}-item__header`;
    const isEnterOrSpace = keyCode === KeyCode.ENTER || keyCode === KeyCode.SPACE;

    if (isEnterOrSpace) {
      if (hasClass(e.target, headerClass)) {
        this.handleItemClick(e);
        return this.props.onPanelHeaderKeyDown(e, keyCode);
      }
    }

    return this.props.onPanelHeaderKeyDown(e, keyCode);
  }

  renderDefaultHeader() {
    const { Header } = this.props;

    return (
      <div onClick={this.handleHeaderClick}>
        {Header}
        <i
          className="icon icon-chevron-down"
          onClick={this.handleItemClick}
        />
      </div>
    );
  }

  render() {
    const {
      prefixCls,
      Header,
      className,
      content,
      isActive,
      disabled,
      children,
      ariaLableExpand,
      ariaLableCollapse,
      isPanelHeaderFocusable
    } = this.props;

    const panelCls = classNames({
      [`${prefixCls}-item`]: true,
      'is-expanded': isActive
    }, className);

    const headerCls = classNames({
      [`${prefixCls}-item__header`]: true,
      'is-disabled': disabled,
      'is-focusable': isPanelHeaderFocusable
    });
    return (
      <div className={panelCls}>
        <a
          className={headerCls}
          aria-disabled={disabled}
          tabIndex={disabled ? -1 : 0}
          href="javascript:void(0)" // eslint-disable-line
          onClick={this.handleItemClick}
          onKeyDown={this.onPanelHeaderKeyDown}
          aria-label={isActive ? ariaLableExpand : ariaLableCollapse}
          role="button"
          aria-expanded={isActive}
        >
          {
            isFunction(Header) ?
              <Header onToggleClick={this.handleItemClick} />
              :
              this.renderDefaultHeader()
          }
        </a>

        <div
          className={`${prefixCls}-item__content`}
        >
          {children || content}
        </div>
      </div>
    );
  }
}

export default Panel;
