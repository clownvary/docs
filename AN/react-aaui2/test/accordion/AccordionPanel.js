import React from 'react'
import { shallow, mount } from 'enzyme'
import sinon from 'sinon'
import renderer from 'react-test-renderer'

import Panel from 'accordion/AccordionPanel'

const props = {
  onInitChildren: jest.fn(),
  setStatus: jest.fn(),
  index: 1,
  transition: 'height 0.3s ease',
  height: 30,
  active: true,
  title: 'test title',
  disabled: false,
  complete: false,
}

it('should render without errors', () => {
  const wrapper = shallow(<Panel {...props}>Test</Panel>)

  expect(wrapper).toBeTruthy()
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer.create(<Panel {...props}>Test</Panel>).toJSON()

  expect(tree).toMatchSnapshot()
})

it('should render self defined class name', () => {
  const wrapper = mount(
    <Panel {...props} className="test">
      Test
    </Panel>,
  )

  expect(wrapper.hasClass('test')).toBe(true)
})

it('should render right with disabled', () => {
  const wrapper = mount(
    <Panel {...props} disabled>
      Test
    </Panel>,
  )

  expect(wrapper.find('.accordion--disabled').length).toBe(1)
})

it('should render right with complete', () => {
  const wrapper = mount(
    <Panel {...props} complete>
      Test
    </Panel>,
  )

  expect(wrapper.find('.accordion--complete').length).toBe(1)
})

it('should render right with no active', () => {
  const wrapper = mount(
    <Panel {...props} active={false}>
      Test
    </Panel>,
  )

  expect(wrapper.find('.icon-plus').length).toBe(1)
})

it('should trigger onInit and onClick', () => {
  const clickFun = jest.fn()
  const wrapper = mount(
    <Panel {...props} transition="height 0.2s ease" setStatus={clickFun}>
      Test
    </Panel>,
  )

  wrapper.find('.accordion__header-icon a').simulate('click')
  expect(props.onInitChildren).toBeCalled()
  expect(clickFun).toBeCalledWith(props.index, false)
})

it('should not trigger onInit and onClick when disabled', () => {
  const clickFun = sinon.spy()
  const wrapper = mount(
    <Panel {...props} disabled setStatus={clickFun}>
      Test
    </Panel>,
  )

  wrapper.find('.accordion__header-icon a').simulate('click')
  expect(clickFun.callCount).toBe(0)
})
