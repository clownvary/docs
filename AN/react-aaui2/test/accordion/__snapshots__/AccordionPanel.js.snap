// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<div
  className="accordion accordion--show   "
>
  <div
    className="accordion__header"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    <div
      className="accordion__header-text"
    >
      test title
    </div>
    <div
      className="accordion__header-icon"
    >
      <a
        onClick={[Function]}
      >
        <i
          className="icon-minus"
        />
      </a>
    </div>
  </div>
  <div
    className="accordion__body-container"
    onTransitionEnd={[Function]}
    style={
      Object {
        "height": undefined,
        "overflow": "hidden",
        "transition": "height 0.3s ease",
      }
    }
  >
    <div
      className="accordion__body"
    >
      Test
    </div>
  </div>
</div>
`;
